helix-term/src/commands.rs

```diff
@@ -5115,21 +5115,26 @@ fn surround_replace(cx: &mut Context) {
             }
         };
 
+        let original_selection = selection.clone();
+        let ranges: SmallVec<_> = change_pos.iter().map(|&p| Range::new(p, p + 1)).collect();
+        doc.set_selection(view.id, Selection::new(ranges, 0));
+
         cx.on_next_key(move |cx, event| {
             let (view, doc) = current!(cx.editor);
             let to = match event.char() {
                 Some(to) => to,
-                None => return,
+                None => return doc.set_selection(view.id, original_selection),
             };
             let (open, close) = surround::get_pair(to);
-            let transaction = Transaction::change(
-                doc.text(),
-                change_pos.iter().enumerate().map(|(i, &pos)| {
+            let mut i = 0;
+            let transaction =
+                Transaction::change_by_selection(doc.text(), doc.selection(view.id), |range| {
                     let mut t = Tendril::new();
                     t.push(if i % 2 == 0 { open } else { close });
-                    (pos, pos + 1, Some(t))
-                }),
-            );
+                    i += 1;
+                    (range.from(), range.to(), Some(t))
+                });
```

Comment:
Why is the change here necessary? We could just keep the original version that uses enumerate()

Reply:
I thought it would make things more consistent to follow, since the selection has already been updated to show the affected surround positions, the transaction just needs to apply the change function to each selected position.